### ISSUE TITLE
hydrolysis doesn't always find pseudo elements

### ISSUE THREAD

**Issue by rictic** [Polymer/polymer-analyzer#187]
``` bash
$ cat example.html <
<script>
  /**
   * Test.
   *
   * @pseudoElement pseudo-elem
   */
</script>


$ cat example.js <
var hydrolysis = require('hydrolysis');

hydrolysis.Analyzer.analyze('example.html').then(function(data) {
  console.log('Elements found:');
  console.log(data.elements);
});


$ node example.js
Elements found:
[]
```

**Comment by rictic** [comment]
Update: AJ discovered that the issue was that pseudo-elements are only found in HTML comments, not jsdoc.

**Comment by rictic** [comment]
We should find them in both HTML comments and JS comments. The analyzer appears not to be finding either, so it doesn't find iron-icons. Or as @MaKleSoft said on #377:

> [...] element descriptions for elements that aren't actually registered anywhere via the `@pseudoElement` tag (hence 'pseudo elements'). `iron-icons` uses this  for example: https://github.com/PolymerElements/iron-icons/blob/master/iron-icons.html#L36

**Comment by MaKleSoft** [comment]
@rictic Actually, #409 only takes care of HTML comments, so we might wanna keep this open for now. (I'll create a PR for JS comments, soon)

**Comment by justinfagnani** [comment]
You know, pseudo element is a really bad name for this, since that's already a thing in HTML/CSS. I'd  expect a pseudo element scanner to be looking for `foo::bar` CSS selectors. I wonder if we can migrate to something else over time.

**Comment by rictic** [comment]
Yeah, agreed. If we can agree on a better name we could recognize both in the analyzer but emit a Warning if you use the old style.

Brainstorming for terminology 🚲🏠ing for this class of thing generally (as we'd like the ability to have pseudo behaviors, pseudo properties, etc):
  * unknown implementation
  * declared

**Comment by justinfagnani** [comment]
Why not just `@element`?

**Comment by rictic** [comment]
That's not bad.

@MaKleSoft has a use case that involves distinguishing these elements so he's adding a `pseudo-element` kind to the generated ScannedPolymerElement but I think that might be better as `implementation-unknown`

**Comment by MaKleSoft** [comment]
Absolutely agree that 'pseudo-element' is a terrible name for this, especially considering that it's never actually used to describe an element of any kind but rather things like element bundles, icon sets, style modules etc. My guess is that the name was chosen historically because the documentation format was derived from that of elements and probably the same logic was used to generate the documentation. This was probably done for convenience but if you think about it, most of the descriptors used to document an element don't really apply to the things we're using `@pseudoElement` for today. Going forward, I think it would be useful to introduce the concept of a separate entity that describes something like a module rather than an element. 'module' is just as ambiguous as 'pseudo element' though. How about 'package', or 'component-package`? Packages could have different types with an optional set of type-specific descriptors. I'm envisioning something like this:

### Element bundle
```
@package paper-elements
@type element-bundle
@elements paper-button, paper-checkbox ...
@main paper-elements.html
```

### Icon set
```
@package iron-icons
@type icon-set
```

### Style module
```
@package paper-styles
@type style-module
```

What do you think?
